### ISSUE TITLE
Fetch does not throw an error in case of a >= 300 status code.

### ISSUE THREAD

**Issue by rishabhpoddar** [supertokens/supertokens-auth-react#31]
This might affect how error handling of queries to the API behaves.

To see `fetch` behaviour, please run the following on any browser:
```js
fetch("https://httpstat.us/500")
    .then(function(response) {
        console.log("error not thrown", response.status);
    })
```
And the output of the above will be "error not thrown 500"

**Comment by NkxxkN** [comment]
Good catch! 

We are already throwing an error for anything that's above 300:  https://github.com/supertokens/supertokens-auth-react/blob/0.0/lib/ts/httpRequest.ts#L80

We only need to make sure to let our users know that we expect them to throw an error in the docs section about overwrritting the props for API Callbacks.

**Comment by rishabhpoddar** [comment]
Cool. Though, now you don't need the enclosing try catch. Since you are throwing an error, which will get caught by that, which will again throw a new error.. it doesn't make any diff from a performance / code flow point of view, but from a code quality view, it's a little strange.

**Comment by NkxxkN** [comment]
You are right. Will fix in `0.0-review` branch.
